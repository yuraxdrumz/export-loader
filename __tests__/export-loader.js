process.env.DEBUG = 'parser'
const { generateExports } = require('../src/esprima')
const exportLoader = require('../src/export-loader')
const fs = require('fs');
const esprima = require('esprima');
const path = require('path')

const data = `
const Public = 'awsome'
const notPublic = function(){
  console.log('this is private because the name is not capitalized!')
}
function SoPublic(){
  console.log('blaaa')
}
`

describe('general tests for main export-loader func and esprima(the parser) func',()=>{
  test('should return func when called', () => {
    expect(generateExports(esprima)).toBeInstanceOf(Function)
  });

  test('should throw error when data not passed', () => {
    expect(()=>generateExports(esprima)()).toThrowError('No data was received!')
  });

  test('should return object with filename, fields to be exported and string of export with fields to add to file content', () => {
    expect(generateExports(esprima)(__filename, data)).toMatchSnapshot()
  });

  test('should return content passed with exports on it', () => {
    expect(exportLoader(data)).toMatchSnapshot()
  });
})
