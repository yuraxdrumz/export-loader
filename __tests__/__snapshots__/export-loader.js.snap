// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`general tests for main export-loader func and esprima(the parser) func should return content passed with exports on it 1`] = `
"
const Public = 'awsome'
const notPublic = function(){
  console.log('this is private because the name is not capitalized!')
}
function SoPublic(){
  console.log('blaaa')
}

export {
d,C,bla,ggg,Utilizer,Public,B,D,srivate,Cat
}"
`;

exports[`general tests for main export-loader func and esprima(the parser) func should return object with filename, fields to be exported and string of export with fields to add to file content 1`] = `
Object {
  "fileName": "/home/yurik/Desktop/export-loader/__tests__/export-loader.js",
  "finalExports": "export {
Public,SoPublic
}",
  "names": Array [
    "Public",
    "SoPublic",
  ],
}
`;
